### ISSUE TITLE
The user_id property for the MessageProperties class appears limited to 8 characters across the wire

### ISSUE THREAD

**Issue by digimaun** [Azure/azure-uamqp-python#125]
For some reason sending user_id with more than 8 characters causes the exceeding characters to get truncated.

For example sending a payload with:

msg_props.user_id = "abcdefghijklmnop/"

On the receiving end the user_id is "abcdefgh"

**Comment by yunhaoling** [comment]
Hi @digimaun , thanks for report.

Which service are you using, it's IoTHub or EventHub or ServiceBus?

**Comment by digimaun** [comment]
This is against IoT Hub. I doubt its a service limitation because for example the C# SDK supports far greater user Id length.

**Comment by yunhaoling** [comment]
Hello @digimaun ,

I confirmed the issue, after sending a message with the message properties being `"testuserid", and when I received the message, it only keeps the first 8 chars, and suffix are weird chars(like Unicode):

b'testuser\x99\xebs\xc8'
b'testuser)\x11\x07\xbe\x82\x18'
b'testuser\xea,\xe92'
b'testuser\xb3\xa1\x9d\xab'

Will investigate the root cause for this

**Comment by yunhaoling** [comment]
hey @digimaun, I'm sorry it took me so long to finally locate the issue.

The root cause of this issue is that the `length` of `user_id` in our cython layer is fixed to the size of a `char*` which is a pointer, so it's always 8 as it's a memory address.

it's in `src/roperties.pyx`:
```python
def user_id(self, char* value):
    cdef c_amqpvalue.amqp_binary _binary
    length = sizeof(value)  # this line is the root cause for the fixed length
    _binary.length = length
    _binary.bytes = value
        # ...
```

we have released uamqp v1.2.13 to address the issue, please upgrade to the latest version.
thanks for your patience!
